RecoMuon/MuonIdentification/plugins/ME0SegmentMatcher.cc

```diff
@@ -164,6 +164,8 @@ void ME0SegmentMatcher::produce(edm::Event& ev, const edm::EventSetup& setup) {
 
       //Remove later
       if (std::abs(thisTrack->eta()) < 1.8) continue;
+      // low pt tracks cannot be propagated to me0
+      if (thisTrack->pt() < 1.0) continue;
```

Comment:
Hi Jason,

``` |pz|>2.5 || p > 3 ``` may be a more inclusive selection to expect the propagation to succeed.
You should anyways add a check  ```  lastrecostate.isValid  ``` after the propagation before using lastrecostate.